File: testcases/test1.java
import java.util.HashMap;

public class TwoSum {
    /**
     * Finds two numbers in the array `nums` that add up to the `target`.
     * Returns their indices.
     *
     * @param nums Array of integers
     * @param target Target sum
     * @return Array of two indices
     */
    public static int[] twoSum(int[] nums, int target) {
        // Map to store numbers and their indices
        HashMap<Integer, Integer> numMap = new HashMap<>(); 
        for (int i = 0; i < nums.length; i++) {
            // Find the complement
            int complement = target - nums[i]; 
            if (numMap.containsKey(complement)) { // Check if complement exists in the map
                return new int[] { numMap.get(complement), i }; // Return the indices
            }
            // Store the current number and its index
            numMap.put(nums[i], i); 
        }
        // Return empty array if no solution is found
        return new int[] {}; 
    }

    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        int target = 9;
        int[] result = twoSum(nums, target);
        System.out.println("Indices: [" + result[0] + ", " + result[1] + "]"); // Output: [0, 1]
    }
}

File: testcases/test1.c
#include <stdio.h>
#include <stdlib.h>

int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    /**
     * Finds two numbers in the array `nums` that add up to the `target`.
     * Returns their indices.
     *
     * @param nums Array of integers
     * @param numsSize Size of the array
     * @param target Target sum
     * @param returnSize Pointer to return the size of the result array
     * @return Array of two indices
     */
    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            // Check if the sum matches the target
            if (nums[i] + nums[j] == target) { 
                int* result = (int*)malloc(2 * sizeof(int));
                result[0] = i;
                result[1] = j;
                *returnSize = 2;
                // Return the indices
                return result; 
            }
        }
    }
    // Return size as 0 if no solution is found
    *returnSize = 0; 
    return NULL;
}

int main() {
    int nums[] = {2, 7, 11, 15};
    int target = 9;
    int returnSize;
    int* result = twoSum(nums, 4, target, &returnSize);
    if (returnSize == 2) {
        printf("Indices: [%d, %d]\n", result[0], result[1]); // Output: [0, 1]
        free(result);
    } else {
        printf("No solution found.\n");
    }
    return 0;
}
